libsolidity/analysis/ControlFlowGraph.cpp

```diff
@@ -34,15 +34,34 @@ bool CFG::constructFlow(ASTNode const& _astRoot)
 
 bool CFG::visit(FunctionDefinition const& _function)
 {
-	if (_function.isImplemented())
-		m_functionControlFlow[&_function] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function);
+	if (_function.isImplemented() && _function.isFree())
+		m_functionControlFlow[pair(nullptr, &_function)] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function);
```

Comment:
```suggestion
		m_functionControlFlow[{nullptr, &_function}] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function);
```

Also, I think `tuple` is generally recommended over `pair` in modern code.